core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -42,9 +42,10 @@ type OutgoingConnectorHandler struct {
 	incomingRateLimiter *common.RateLimiter
 	outgoingRateLimiter *common.RateLimiter
 	responses           *responses
+	selectorOpts        []func(*RoundRobinSelector)
 }
 
-func NewOutgoingConnectorHandler(gc connector.GatewayConnector, config ServiceConfig, method string, lgger logger.Logger) (*OutgoingConnectorHandler, error) {
+func NewOutgoingConnectorHandler(gc connector.GatewayConnector, config ServiceConfig, method string, lgger logger.Logger, opts ...func(*RoundRobinSelector)) (*OutgoingConnectorHandler, error) {
```

Comment:
@MStreet3 Any reason why we wouldn't flip the default around? i.e. by default we make it random, and only if you pass in a non-random flag will we make it deterministic?

This ensures that by default we'll have the right behaviour while preserving the ability to override in tests